jolt-core/src/subprotocols/optimization.rs

```diff
@@ -0,0 +1,761 @@
+                                cur.0 * mle_vec[D - 2 - i].get_bound_coeff(j),
+                                cur.1
+                                    * mle_vec[D - 2 - i]
+                                        .get_bound_coeff(j + mle_vec[D - 2 - i].len() / 2),
+                            );
+                        }
+                    }
+                });
+
+            drop(_guard);
+            drop(inner_span);
+
+            let inner_span = tracing::span!(tracing::Level::INFO, "Loop over d");
+            let _guard = inner_span.enter();
+
+            for d in 0..D {
+                let round = j_idx * D + d;
+
+                if d == 0 {
+                    if round > 0 {
+                        C *= C_summands[0] + C_summands[1];
+                    }
+
+                    let r_cycle_val = r_cycle[j_idx];
+
+                    C_summands[0] = r_cycle_val;
+                    C_summands[1] = F::one() - r_cycle_val;
+                }
+
+                // Compute eq(r_round, w_1, ..., w_{idx - 1}, c, b) for each c and b = 0, 1
+                // let eq_evals_at_idx = eval_points
+                //     .iter()
+                //     .map(|c| (((F::one() - c) * C_summands[1], *c * C_summands[0])))
+                //     .collect::<Vec<(F, F)>>();
+
+                // Evaluate eq(r_round, w_1, ..., w_{idx - 1}, c) at c = 0, 2
+                let eq_evals_at_idx = [
+                    (C_summands[1], F::zero()),
+                    (-C_summands[1], C_summands[0] + C_summands[0]),
+                ];
+
+                let univariate_poly_evals = before_idx_evals
+                    .par_iter_mut()
+                    .take(size)
+                    .zip(after_idx_evals.par_iter_mut().take(size))
+                    .enumerate()
+                    .map(|(j, (before_idx_eval, after_idx_evals))| {
+                        // let at_idx_evals =
+                        // mle_vec[D - d - 1].sumcheck_evals(j, 2, BindingOrder::HighToLow);
+                        let at_idx_evals = [
+                            mle_vec[D - d - 1].get_bound_coeff(j),
+                            mle_vec[D - d - 1].get_bound_coeff(j + mle_vec[D - d - 1].len() / 2)
+                                + mle_vec[D - d - 1]
+                                    .get_bound_coeff(j + mle_vec[D - d - 1].len() / 2)
+                                - mle_vec[D - d - 1].get_bound_coeff(j),
+                        ];
+
+                        let eq_eval_after_idx = if j_idx < T.log_2() - 1 {
+                            E_table[j_idx].get_coeff(j)
+                        } else {
+                            F::one()
+                        };
+
+                        if d > 0 {
+                            *before_idx_eval = before_idx_eval
+                                .mul_1_optimized(mle_vec[D - d - 1].get_bound_coeff(j));
+                        }
+
+                        let temp = *before_idx_eval * eq_eval_after_idx * C;
```

Comment:
Since `before_idx_eval` will always be 1 for d = 0
```suggestion
                        let temp = before_idx_eval.mul_1_optimized(eq_eval_after_idx) * C;
```